### ISSUE TITLE
Are we missing some Importiant Maths Operators in 5.1.20?

### ISSUE THREAD

**Issue by AnthonyMuscio** [Jermolene/TiddlyWiki5#4128]
I just shared a code pattern that includes a case structure in the [forum](https://groups.google.com/forum/?hl=en#!topic/tiddlywiki/wGP5yGWoKgk)
My example case structure is based on a set of fixed values.

You will see at the end I suggest how it will work nicely with the new mathematics operators, and whilst the list widgets could be exchanged for reveal widgets it makes me think we could have  greater than, lessthan and equal operators. These could be just for numbers or extended to handle string comparisons as well.

Sure Joshuas republishing evans formulas plugin will help. But this seems a small step to make.

These could be used two ways
`[<value>!equal[5]]`
or
`1 2 3 4 5 6 7 8 9 +[greaterthan[5]]`

What do people think?
Regards
Tony

**Comment by Jermolene** [comment]
Hi @AnthonyMuscio there's actually a few tickets/PRs covering this already: #3781, #2253, #2968 and #3760. Perhaps the big question now is whether we want to defer v5.1.20 long enough to get one of these solutions in. Right now v5.1.20 is waiting on me fixing the regression in #4082

**Comment by AnthonyMuscio** [comment]
Jeremy, your call of course.

I think the equals operator has precidence over greater and less than because it allows the result of maths operators to return a true or false condition. If it can also compare strings it will add a missing feature to filters, a positive test and help make some filters more readable.

With the run operators - + and ~ now = 
we still cant ask if the result equals x then select

[<result>] +[equals[x]]

At present we can only test it is not

[<result>] -[[x]]

I can explain this further if requested.

regards
Tony

**Comment by AnthonyMuscio** [comment]
After any responses here I will close, update existing or create a new issue as this issue is getting a little messy.

**Alternative to Equals**
Perhaps a simple string comparison that does not take account of maths would be a good start.

For example +[sameas[1.2]] would only be true if the result is exactly 1.2
Alternatively we could call this the match operator +[match[1.2]] in keeping with the reveal widget.

 ie it would not handle the mathematics that says 1.20 is equal to 1.2 however **this would be immediately practical for string tests such** as +[sameas[true]] and prefix[1.2] can be used.

**Alternative to greater than and less than etc... operators**
Currently the reveal widget handles this kind of comparison however it does not permit the test to be done against a variable, or replaceable parameter. If the reveal widget was enhanced this way it would at least enable such comparisons 

**Equals operator later**
This would allow time to develop a real equals Operator, perhaps even one that can enclose further mathematics
eg +[equals[2x3.14]] for equals 2 pi

**Comment by AnthonyMuscio** [comment]
Also https://github.com/Jermolene/TiddlyWiki5/issues/2253 and https://github.com/Jermolene/TiddlyWiki5/pull/3760

**Comment by AnthonyMuscio** [comment]
I will close this with the advent of the match Operator and raise the idea of a greaterthan and lessthan operator in a new issue at some time in the future. Thanks soo much for the match operator.
